trk/src/Screens/GymDaily/index.js

```diff
@@ -1,8 +1,219 @@
+
+          latestAscentsCount += taps.filter(tap => tap.timestamp > oneDayAgo).length;
+
+          taps.forEach(tap => {
+            if (!climbFrequencyMap[tap.climb]) {
+              climbFrequencyMap[tap.climb] = { count: 0, name: climb.name, id: climb.id, grade: climb.grade };
+            }
+            climbFrequencyMap[tap.climb].count++;
+          });
+        }
+      }
+
+      for (const tap of allTaps) {
+        if (tap.timestamp) {
+          const hour = tap.timestamp.getHours(); 
+          timeFrequencyMap[hour] = (timeFrequencyMap[hour] || 0) + 1;
+        }
+      }
+
+
+      let peakHour = Object.keys(timeFrequencyMap).reduce((a, b) => timeFrequencyMap[a] > timeFrequencyMap[b] ? a : b, null);
+      setPeakTime(peakHour ? `${peakHour}:00 - ${parseInt(peakHour) + 1}:00` : '');
+      console.log(`peakHour is: ${peakHour}`)
+
+      setAscents(allTaps);
+      setTotalAscents(totalTaps.toString());
+      setLatestAscents(latestAscentsCount.toString());
+
+      let mostCompletedClimb = Object.values(climbFrequencyMap).reduce((a, b) => (a && b && a.count > b.count) ? a : b, null);
+      let leastCompletedClimb = Object.values(climbFrequencyMap).reduce((a, b) => (a && b && a.count < b.count) ? a : b, null);
+
+      if (mostCompletedClimb) {
+        setMostCompleted({ name: mostCompletedClimb.name, id: mostCompletedClimb.id, grade: mostCompletedClimb.grade });
+      } else {
+        setMostCompleted({ name: '', id: '', grade: '' });
+      }
+
+      if (leastCompletedClimb) {
+        setLeastCompleted({ name: leastCompletedClimb.name, id: leastCompletedClimb.id, grade: leastCompletedClimb.grade });
+      } else {
+        setLeastCompleted({ name: '', id: '', grade: '' });
+      }
+
+    };
+
+
+    if (yourClimbs.length > 0) {
+      fetchTapsAndCalculateAscents();
+    }
+  }, [yourClimbs]);
+
+
+  useEffect(() => {
+    const fetchCommentsAndCalculateFeedback = async () => {
+      try {
+        const commentsPromises = yourClimbs.map(climb =>
+          getCommentsBySomeField('climb', climb.id)
+        );
+        const commentsSnapshots = await Promise.all(commentsPromises);
+        let comments = [];
+        commentsSnapshots.forEach(snapshot => {
+          snapshot.docs.forEach(doc => comments.push({ ...doc.data(), timestamp: doc.data().timestamp.toDate() }));
+        });
+
+        setYourComments(comments);
+ 
+        if (comments.length > 0) {
+          const latestComment = comments.reduce((latest, current) =>
+            latest.timestamp > current.timestamp ? latest : current
+          );
+
+          const climbDetails = await getClimb(latestComment.climb);
+          const climbData = climbDetails.data();
+          setLatestFeedback({
+            explanation: latestComment.explanation,
+            climb: latestComment.climb,
+            rating: latestComment.rating,
+            climbName: climbData.name,
+            climbGrade: climbData.grade,
+          });
+        }
+
```

Comment:
when we set latest feedback, we save data about the climb associated to the comment to be able to render it in the view. Also, this will allow in a later PR to make these data boxes clickable and to prop the climbId (which is the climb key value pair on line 168) to show climbDetail